lib/decision_review/service.rb

```diff
@@ -22,61 +18,90 @@ class Service < Common::Client::Base
 
     STATSD_KEY_PREFIX = 'api.decision_review'
 
+    HLR_CREATE_RESPONSE_SCHEMA = VetsJsonSchema::SCHEMAS['HLR-CREATE-RESPONSE-200']
+    raise unless HLR_CREATE_RESPONSE_SCHEMA.is_a? Hash
```

Comment:
This is concerning because wouldn't it raise this exception when the DecisionReview Service is loaded?  It seems like it's checking that VetsJsonSchema has these required values and I think runtime is the wrong time to alert on that mismatch.  I think it would better serve as a test, so that way the pipeline would break if somebody tried to introduce a VetsJsonSchema that is missing these required schemas.  The test would also be more descriptive than this generic exception.

Reply:
tests would fail too, fwiw.
if these values aren't present, an exception will be thrown _somewhere_, and, if that's the situation, might as well make it obvious (hence raise)

Reply:
I think the problem is fundamentally that raising an exception here is this is testing a dependency.  Truly there should just be a test in VetsJsonSchema and we wouldn't need a test here.  But if this dependency needs to be tested, I think the test belongs in the test suite.
You are right though, that if these values come back nil the error message is enigmatic.  But again, I think a test solves this in that the description/context/name will be specific about what's wrong.

Reply:
>should just be a test in VetsJsonSchema and we wouldn't need a test here

i don't believe that's the case. this is catching an integration problem. a test might be passing in vets-json-schema, but, if vets-api is pointing to the wrong version of vets-json-schema (for instance), a schema could be missing (this has already happened to me a couple times)

Reply:
Wouldn't the missing schema issue be related to having an old version of VetsJsonSchema?  There's no risk of advancing the schema and this disappearing, right?
I would still argue that integration is a category of testing, and would prefer an explicit test over a RuntimeError

Reply:
>There's no risk of advancing the schema and this disappearing, right?

not true. I made a schema go bye-bye recently (rename)

Reply:
what about a test _and_ what I have above?

scenario:

a schema gets renamed in vets-json-schema.
i think i've changed all instances of the name (including in my tests which specifically test for those schemas' presence)

i miss it in my controller.
now i'm getting a really strange backtrace.
15 min later i realize it's because i missed a rename

^ this happened. if a schema is nil and gets passed into JsonSchemer, the backtrace is super confusing (i might submit a PR to JsonSchemer actually. it looked like an oversight)

i could have used `fetch` and accomplished roughly the same thing, but i thought it'd be nice to ensure that it was a hash too

Reply:
```
    context 'schema constants' do
      it 'has HLR_CREATE_RESPONSE_SCHEMA hash' do
        expect(described_class::HLR_CREATE_RESPONSE_SCHEMA).to be_a Hash
      end
      it 'has HLR_SHOW_RESPONSE_SCHEMA hash' do
        expect(described_class::HLR_SHOW_RESPONSE_SCHEMA).to be_a Hash
      end
      it 'has HLR_GET_CONTESTABLE_ISSUES_RESPONSE_SCHEMA hash' do
        expect(described_class::HLR_GET_CONTESTABLE_ISSUES_RESPONSE_SCHEMA).to be_a Hash
      end
    end
```
I put these tests in `spec/lib/decision_review/service_spec.rb` and then set `HLR_SHOW_RESPONSE_SCHEMA` to nil and got two failures.  One the enigmatic one from JsonSchemer, but also one that makes it clear that there's an issue with the schema constant.
```
  1) DecisionReview::Service#get_higher_level_review 200 response returns a properly formatted 200 response
     Failure/Error: errors = JSONSchemer.schema(schema).validate(json).to_a
  2) DecisionReview::Service#create_higher_level_review schema constants has HLR_SHOW_RESPONSE_SCHEMA hash
     Failure/Error: expect(described_class::HLR_SHOW_RESPONSE_SCHEMA).to be_a Hash
       expected nil to be a kind of Hash
```

I think these tests give us the clear message that those constants have a bad value, which is unexpected/unacceptable, and would point the developer to the right point in the code.  They aren't susceptible to schema name changes, because we're checking the constant.  And if they rename the constant those tests would also fail because the constants name was changed.
My main concern about raising an exception in the main file here is that it fails when the file is loading, which would break the vets-api deploy. I think the right way to make sure changes aren't breaking features is through a test.

Reply:
i suppose i don't necessarily want to check the constants defined in the controller, I want to check the contents of the hash `VetsJsonSchema::SCHEMAS`

Reply:
made some updates! let me know 👍

Reply:
Much better!  Thanks for looking at this again!